IPython/core/completer.py

```diff
@@ -1984,11 +1990,15 @@ def _complete(self, *, cursor_line, cursor_pos, line_buffer=None, text=None,
                 return latex_text, latex_matches, ['latex_matches']*len(latex_matches), ()
             name_text = ''
             name_matches = []
+            # need to add self.fwd_unicode_match() function here when done
             for meth in (self.unicode_name_matches, back_latex_name_matches, back_unicode_name_matches):
                 name_text, name_matches = meth(base_text)
+                completion_text, completion_matches = self.fwd_unicode_match(base_text)
                 if name_text:
                     return name_text, name_matches[:MATCHES_LIMIT], \
                            [meth.__qualname__]*min(len(name_matches), MATCHES_LIMIT), ()
+                elif (completion_text):
+                    return completion_text, completion_matches, ()
```

Comment:
those 3 things are not quite correct. Python have functions and methods as first class values, so you can do 

```diff
- for meth in (self.unicode_name_matches, back_latex_name_matches, back_unicode_name_matches)
+ for meth in (self.unicode_name_matches, back_latex_name_matches, back_unicode_name_matches, self.fwd_unicode_match)
```

And it will work better.